src/base/JSON.ml

```diff
@@ -167,7 +167,7 @@ and read_adt_json name j tlist_verify =
   let verify_exn name tlist1 adt =
     match adt with
     | ADTValue (_, tlist2, _) ->
-        if type_equiv_list tlist1 tlist2 then ()
```

Comment:
Well spotted.

I have taken the liberty of removing the wildcard pattern at the end of the pattern match in `json_to_lit` and `build_prim_lit_exn` functions. I really don't like using these sorts of catchall patterns, because they make it more difficult to spot issues like this when a type is extended or changed.